db-tests/src/test/java/TestTriggers.java

```diff
@@ -92,4 +100,438 @@ void afterall() throws SQLException, IOException, InterruptedException {
     proc.waitFor();
     proc.destroy();
   }
+
+  int insertFileUpload() throws SQLException {
+    final var res = connection.createStatement()
+                              .executeQuery(
+                                  """
+                                      INSERT INTO uploaded_file (path, name)
+                                      VALUES ('test-path', 'test-name-%s')
+                                      RETURNING id;"""
+                                      .formatted(UUID.randomUUID().toString())
+                              );
+    res.next();
+    return res.getInt("id");
```

Comment:
I'm getting a bunch of these warnings when I run these tests:

<img width="791" alt="Screen Shot 2022-01-05 at 4 02 46 PM" src="https://user-images.githubusercontent.com/1189602/148307119-f4f66408-5243-433a-a7c8-7c82d2780609.png">

What I think is happening is each statement needs to be "cleaned up" when we're done iterating through its results (I think). The typical way to do this is to use a "try-with-resources", something along these lines:

```suggestion
    try (final var statement = connection.createStatement()) {
      final var res = statement.executeQuery(
          """
              INSERT INTO uploaded_file (path, name)
              VALUES ('test-path', 'test-name-%s')
              RETURNING id;"""
              .formatted(UUID.randomUUID().toString())
      );
      res.next();
      return res.getInt("id");
    }
```

Reply:
I updated to use the try-with-resources where possible, and then explicitly calling close in other cases. Can you try again and see how it goes?

Reply:
That worked! The only warnings I see now are these, which I think are okay:
<img width="829" alt="Screen Shot 2022-01-06 at 1 15 36 PM" src="https://user-images.githubusercontent.com/1189602/148453363-a509b371-8d71-474a-a855-b7b94abc6bcd.png">

Reply:
Just so you can reproduce it - I'm going into TestTrigger.java and clicking the green play button next to "DatabaseTests", then clicking around in the test results pane below.

Reply:
I'm running with `./gradlew test`

For some reason intellij isn't detecting the test to be run alone